### ISSUE TITLE
[6.*] output path for PostCSS seems to be buggy

### ISSUE THREAD

**Issue by mohamadnorouzi20** [JeffreyWay/laravel-mix#2517]
-   Laravel Mix Version: 6.0.0-beta.4
-   Node Version: v12.18.3
-   NPM Version: 6.14.8
-   OS: Fedora 32

### Description:
This used to work just fine in Mix 5 (pay attention to `build` directory and also the `.pcss` extension)
```js
const mix = require('laravel-mix')

mix.js('resources/js/app.js', 'public/build/js')
mix.postCss('resources/pcss/app.pcss', 'public/build/css', [
    //
])
```

But now I'm getting this after running `npx mix build`:
![image](https://user-images.githubusercontent.com/8435784/95107982-7e6bd280-0747-11eb-8e75-0ffe6b0ff316.png)

It's dumping the `app.css` file inside the `js` directory

**Comment by mohamadnorouzi20** [comment]
I checked a few things and it seems this only happens because of the `.pcss` extension. it seems like if the extension does not belong to some list of allowed extensions it will ignore the **postCss output path** and use **js output path** instead.

**Comment by JeffreyWay** [comment]
I don't think we've ever supported the `pcss` extension, so I'm confused how it was working in Mix 5.

**Comment by thecrypticace** [comment]
I _think_ this should still place the file in the right location regardless of extension. I'll check on that.

**Comment by cds-mm** [comment]
Experiencing a similar issue using `.sass` except it is producing 2 `.css` files in different directories:
```javascript
const mix = require('laravel-mix')

mix.js('src/app.js', 'public/js').vue({version: 2})
mix.sass('src/app.scss', 'public/css')
```
![image](https://user-images.githubusercontent.com/48589485/95230896-6012df00-07fa-11eb-8887-02a4f6e71790.png)

**Comment by thecrypticace** [comment]
@cds-mm The only way I can get multiple output files like that is to run postCss + sass for separate files with the same output file name. Can you post a minimal reproduction?

**Comment by thecrypticace** [comment]
@mohamadnorouzi20 @JeffreyWay We don't support `.pcss` extensions. We could though. All it would require is changing the `chunkRegex` for postCSS.

**Comment by JeffreyWay** [comment]
This issue might be related: #2522

Something odd does seem to be happening with CSS extraction is certain cases.

**Comment by cds-mm** [comment]
@JeffreyWay @thecrypticace I think this issue and #2522 are very much related. I created a Node project that only uses Laravel-Mix and Bootstrap. My `webpack.mix.js` contents where similar to what they were previously (without using Vue):
```javascript
const mix = require('laravel-mix')

mix.js('src/app.js', 'public/js');
mix.sass('src/app.scss', 'public/css');
```

I then experimented with the contents of the `app.scss` file. When the contents is just `sass` that needs compiling to `css`
```css
$primary: grey;

.app {
    background: $primary;
}
```
The correct files are generated:
![image](https://user-images.githubusercontent.com/48589485/95255804-d541dc80-0819-11eb-84b1-d516052d8d0e.png)

But once I ask it to also import Bootstrap with `sass` or just import Bootstrap by itself, then it creates the extra `.css` file in `public/js`
```css
@import '~bootstrap/dist/css/bootstrap.css';

$primary: grey;

.app {
    background: $primary;
}
```
![image](https://user-images.githubusercontent.com/48589485/95256075-3b2e6400-081a-11eb-8a8d-12922258e425.png)

Additionally what I've noticed is that the contents of `public/js/app.css` is just Bootstrap, while `public/css/app.css` has the rest of my `sass` compiled to `css`.

Hope this helps somewhat in tracking down the cause of the issue.
